components/script/dom/element.rs

```diff
@@ -701,7 +700,9 @@ pub trait AttributeHandlers {
 
 impl<'a> AttributeHandlers for JSRef<'a, Element> {
     fn get_attribute(self, namespace: &Namespace, local_name: &Atom) -> Option<Temporary<Attr>> {
-        self.get_attributes(local_name).into_iter().map(|attr| attr.root())
+        let mut attributes = RootedVec::new();
+        self.get_attributes(local_name, &mut attributes);
+        attributes.iter().map(|attr| attr.root())
```

Comment:
Now that Temporary::root() does not consume the Temporary value, that could just be:

``` rust
attributes.iter().find(|attr| attr.root().r().namespace() == namespace)
```